Windows: compiler flags not forwarded

The test `//tests/binary-with-compiler-flags` fails to build on Windows, the flags are not forwarded:

```
PS C:\Users\nmattia\projects\tweag\rules_haskell> bazel test --config windows //tests/binary-with-compiler-flags
WARNING: The following rc files are no longer being read, please transfer their contents or import their path into one of the standard rc files:
c:\users\nmattia\projects\tweag\rules_haskell/.bazelrc
INFO: Invocation ID: cc0a040f-a0fb-4595-9489-0252dc85228c
INFO: Build options have changed, discarding analysis cache.
INFO: Analysed target //tests/binary-with-compiler-flags:binary-with-compiler-flags (0 packages loaded, 412 targets configured).
INFO: Found 1 test target...
ERROR: C:/users/nmattia/projects/tweag/rules_haskell/tests/binary-with-compiler-flags/BUILD:8:1: error executing shell command: 'C:/Users/nmattia/dadew/scoop/apps/msys-rev13/current/bin/bash.exe -c
        export PATH=${PATH:-} # otherwise GCC fails on Windows

        # this is equivalent to 'readarray'. We do use 'read...' failed (Exit 1)

tests\binary-with-compiler-flags\Main.hs:5:1: error:
    Illegal standalone deriving declaration
      Use StandaloneDeriving to enable this extension
  |
5 | deriving instance Show Foo
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^
Target //tests/binary-with-compiler-flags:binary-with-compiler-flags failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 6.932s, Critical Path: 0.28s
INFO: 0 processes.
FAILED: Build did NOT complete successfully

FAILED: Build did NOT complete successfully
```

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -31,6 +31,7 @@ jobs:
       /c/bazel/bazel.exe test --config windows "//tests/binary-custom-main/..."
       /c/bazel/bazel.exe test --config windows "//tests/binary-with-main/..."
       /c/bazel/bazel.exe test --config windows "//tests/binary-simple/..."
+      /c/bazel/bazel.exe test --config windows "//tests/binary-with-compiler-flags/..."
       /c/bazel/bazel.exe test --config windows "//tests/binary-with-link-flags/..."
       /c/bazel/bazel.exe test --config windows "//tests/encoding/..."
 

